src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -190,10 +216,55 @@ export default {
     // Remove space-character
     processedUserInput: ({ debouncedInput }) => debouncedInput.replace(/\s/g, ''),
     totalItemsToNavigate: ({ filteredSymbols }) => filteredSymbols.length,
+    selectedSymbol: ({
+      filteredSymbols,
+      focusedIndex,
+    }) => (focusedIndex !== null ? filteredSymbols[focusedIndex] : null),
+    nextSymbol: ({
+      filteredSymbols,
+      focusedIndex,
+    }) => {
+      if (focusedIndex === null) {
+        return null;
+      }
+      let nextIndex = focusedIndex + 1;
+      if (nextIndex >= filteredSymbols.length) {
+        nextIndex = 0;
+      }
+      return filteredSymbols[nextIndex];
+    },
+    previewEnabled: () => getSetting(
+      ['features', 'docs', 'quickNavigationPreview', 'enable'],
+      false,
+    ),
+    previewJSON: ({
+      cachedSymbolResults,
+      selectedSymbol,
+    }) => (selectedSymbol ? (
+      (cachedSymbolResults[selectedSymbol.uid] || {}).json
+    ) : (
+      null
+    )),
+    previewState: ({
+      cachedSymbolResults,
+      previewIsLoadingSlowly,
+      selectedSymbol,
+      // eslint-disable-next-line no-nested-ternary
+    }) => (selectedSymbol && Object.hasOwnProperty.call(cachedSymbolResults, selectedSymbol.uid) ? (
```

Comment:
could this be simplified to something like this or that would cause false positives?
```suggestion
    }) => (selectedSymbol && cachedSymbolResults[selectedSymbol.uid] ? (
```

Reply:
Yes but it makes the linter angry 😝

Reply:
Why does it make the linter angry?

Reply:
I don't think it applies to this situation or others in the codebase, but I think this is the one that usually pops up from time to time: https://eslint.org/docs/latest/rules/no-prototype-builtins